src/lib.rs

```diff
@@ -447,15 +456,28 @@ impl DstLayout {
         __debug_assert!(self_bytes <= bytes_len);
 
         let split_at = match cast_type {
-            _CastType::_Prefix => self_bytes,
+            CastType::Prefix => self_bytes,
             // Guaranteed not to underflow because `self_bytes <= bytes_len`
             // (lemma 4).
             #[allow(clippy::arithmetic_side_effects)]
-            _CastType::_Suffix => bytes_len - self_bytes,
+            CastType::_Suffix => bytes_len - self_bytes,
         };
 
         Some((elems, split_at))
     }
+
+    /// Calls `validate_exact_cast` and ensures that all of `bytes_len` are
+    /// used.
+    ///
+    /// Passes all arguments to `validate_cast`, and passes `CastType::Prefix`.
+    /// If the cast would leave any suffix bytes left over,
+    /// `validate_exact_cast` returns `None`.
+    const fn validate_exact_cast(&self, addr: usize, bytes_len: usize) -> Option<usize> {
```

Comment:
Consider renaming this:
- `validate` (as opposed to `is_valid`) implies side effects
- `validate` implies that it _only_ performs validation, but it also computes metadata

Some options brainstormed with @jswrenn:
- `try_cast_metadata` / `try_cast_metadata_no_prefix_suffix`